tfhe/src/shortint/parameters/mod.rs

```diff
@@ -115,6 +115,45 @@ impl ClassicPBSParameters {
             encryption_key_choice,
         }
     }
+
+    pub fn to_shortint_conformance_param(&self) -> CiphertextConformanceParams {
+        let (pbs_order, expected_dim) = match self.encryption_key_choice {
+            EncryptionKeyChoice::Big => (
+                PBSOrder::KeyswitchBootstrap,
+                self.glwe_dimension
+                    .to_equivalent_lwe_dimension(self.polynomial_size),
+            ),
+            EncryptionKeyChoice::Small => (PBSOrder::BootstrapKeyswitch, self.lwe_dimension),
+        };
+
+        let message_modulus = self.message_modulus;
+        let ciphertext_modulus = self.ciphertext_modulus;
+        let carry_modulus = self.carry_modulus;
+
+        let degree = Degree(message_modulus.0 - 1);
+
+        CiphertextConformanceParams {
+            ct_params: LweCiphertextParameters {
+                lwe_dim: expected_dim,
+                ct_modulus: ciphertext_modulus,
+            },
+            message_modulus,
+            carry_modulus,
+            pbs_order,
+            degree,
+        }
+    }
+
+    #[cfg(feature = "integer")]
+    pub fn to_integer_conformance_param(
+        &self,
+        num_blocks_per_integer: usize,
+    ) -> crate::integer::parameters::CiphertextConformanceParams {
+        crate::integer::parameters::CiphertextConformanceParams {
+            shortint_params: self.to_shortint_conformance_param(),
+            num_blocks_per_integer,
+        }
+    }
```

Comment:
can we implement a from for the crate::integer::parameters::CiphertextConformanceParams from the shortint parameter ? I don't like inverting the knowledge requirement for the layers, shortint does not have to know about integer

Reply:
Sure, I did it this way because it's shorter to call but I agree it's not great
I'll change it

Reply:
I understand and it's good to make progress :) but yeah not mergeable I believe